File: backend/models/analytics.go
package models

import (
	"database/sql"
	"time"
)

type AnalyticsType string

const (
	AnalyticsTypePhysics   AnalyticsType = "physics"
	AnalyticsTypeCS        AnalyticsType = "computer_science"
)

type AnalyticsData struct {
	ID          string        `json:"id"`
	Title       string        `json:"title"`
	Description string        `json:"description"`
	DataType    AnalyticsType `json:"data_type"`
	Data        interface{}   `json:"data"`
	CreatedAt   time.Time     `json:"created_at"`
	UpdatedAt   time.Time     `json:"updated_at"`
}

type AnalyticsDataDB struct {
	ID          string       `db:"id"`
	Title       string       `db:"title"`
	Description string       `db:"description"`
	DataType    string       `db:"data_type"`
	Data        []byte       `db:"data"`
	CreatedAt   time.Time    `db:"created_at"`
	UpdatedAt   sql.NullTime `db:"updated_at"`
}

File: frontend/src/types/analytics.ts
export type AnalyticsType = 'physics' | 'computer_science'

export interface AnalyticsData {
  id: string
  title: string
  description: string
  dataType: AnalyticsType
  data: any
  createdAt: Date
  updatedAt: Date
}

export interface PhysicsData {
  measurements?: Array<{
    energy: number
    time: number
  }>
  collisions?: Array<{
    momentum: number
    angle: number
  }>
  units?: {
    energy?: string
    time?: string
    momentum?: string
    angle?: string
  }
  detector?: string
  energy_level?: string
}

export interface ComputerScienceData {
  algorithms?: Array<{
    name: string
    runtime: number[]
    inputSize: number[]
  }>
  metrics?: Array<{
    timestamp: string
    bandwidth: number
    latency: number
  }>
  units?: {
    bandwidth?: string
    latency?: string
  }
}
